presto-main/src/main/java/io/prestosql/sql/planner/PlanOptimizers.java

```diff
@@ -233,30 +234,33 @@ public PlanOptimizers(
         Set<Rule<?>> predicatePushDownRules = ImmutableSet.of(
                 new MergeFilters(metadata));
 
-        Set<Rule<?>> columnPruningRules = ImmutableSet.of(
-                new PruneAggregationColumns(),
-                new PruneAggregationSourceColumns(),
-                new PruneApplyColumns(),
-                new PruneApplySourceColumns(),
-                new PruneAssignUniqueIdColumns(),
-                new PruneCorrelatedJoinColumns(),
-                new PruneDeleteSourceColumns(),
-                new PruneEnforceSingleRowColumns(),
-                new PruneFilterColumns(),
-                new PruneIndexSourceColumns(),
-                new PruneJoinChildrenColumns(),
-                new PruneJoinColumns(),
-                new PruneMarkDistinctColumns(),
-                new PruneOutputColumns(),
-                new PruneProjectColumns(),
-                new PruneSemiJoinColumns(),
-                new PruneSemiJoinFilteringSourceColumns(),
-                new PruneTopNColumns(),
-                new PruneValuesColumns(),
-                new PruneWindowColumns(),
-                new PruneOffsetColumns(),
-                new PruneLimitColumns(),
-                new PruneTableScanColumns());
+        Set<Rule<?>> columnPruningRules = ImmutableSet.<Rule<?>>builder()
+                .add(
+                        new PruneAggregationColumns(),
+                        new PruneAggregationSourceColumns(),
+                        new PruneApplyColumns(),
+                        new PruneApplySourceColumns(),
+                        new PruneAssignUniqueIdColumns(),
+                        new PruneCorrelatedJoinColumns(),
+                        new PruneDeleteSourceColumns(),
+                        new PruneEnforceSingleRowColumns(),
+                        new PruneFilterColumns(),
+                        new PruneIndexSourceColumns(),
+                        new PruneJoinChildrenColumns(),
+                        new PruneJoinColumns(),
+                        new PruneMarkDistinctColumns(),
+                        new PruneOutputColumns(),
+                        new PruneProjectColumns(),
+                        new PruneSemiJoinColumns(),
+                        new PruneSemiJoinFilteringSourceColumns(),
+                        new PruneTopNColumns(),
+                        new PruneValuesColumns(),
+                        new PruneWindowColumns(),
+                        new PruneOffsetColumns(),
+                        new PruneLimitColumns(),
+                        new PruneTableScanColumns())
+                .addAll(PruneSetOperationSourceColumnsRuleSet.rules())
```

Comment:
I'd flatten these in the list above. There's no benefit in grouping them in a separate collection. It also makes them look different and "special" and requires extra mental work to realize they are not.